ui/src/app/job-list/table/table.component.ts

```diff
@@ -200,4 +200,26 @@ export class JobsTableComponent implements OnInit {
       this.jobs.forEach(j => this.selection.select(j));
     }
   }
+
+  updateCheckBoxSelection(clickedJob: QueryJobsResult, event: MouseEvent): void {
+    // if the user has shift-clicked on a job, find the last selected job and
+    // select all jobs between the two
+    if (event.shiftKey && this.selection.selected.length) {
```

Comment:
opt_nit: could invert this condition and return early to avoid indenting everything

Reply:
it would avoid the indenting, but I think the logic is clearer when you're reading the code this way